### ISSUE TITLE
Test fails in Ubuntu 18.04 - Parallel RBF mapping

### ISSUE THREAD

**Issue by martinsaravia** [precice/precice#526]
Hi PreCICErs, during the last 2 days I have been trying to make things work with PETSc-based parallel RBF mappings. Unfortunately, after trying several things, I couldn't. 

My configuration is: 

- Ubuntu 18.04
- MPI 2.1.1-8 (also tried with 3.0 and 4.0.1)
- PETSC (3.6, dev, 3.11), compiled from source. 
- preCICE 1.6.0

The error message I am getting in make test_base is: [test.log](https://github.com/precice/precice/files/3618831/test.log)

The problem is of course PETSC related, I sent emails to PETSC support and they warned me against issues with old versions of OpenMPI and adviced to install new versions  (newer than 3). However, when I do that, I cannot compile precice to use the new OpenMPI version, it seems that it chooses the system OpenMPI (2.1.1-8). 

The PETSC-OpenMPI problem was already reported in issue #20 (https://github.com/precice/tutorials/issues/20) and as per issue #115 (https://github.com/precice/tutorials/issues/20) the solution wold be to compile PETSC with 64 bits indices (--with-64-bit-indices option). Unfortunately, it did not work for me.

I have tried every PETSC with system OpenMPI setting with no success. As said, I have also tried to compile MPI (version 3.0 and 4.0.1) but when I link preCICE to the solvers I get an incompatible MPI error. 

In some cases, the PETSC parallel tests (ex19) fails... so also fails preCICE. However, in cases where PETSC test passes, preCICE tests fails. 

Sorry if I am violating any of the contributing guidelines, this is my first time here. Thanks in advance.

**Comment by martinsaravia** [comment]
As recommended by PETSC support, I have downloaded version 2.1.1 of MPI, compiled without -enable-heterogeneous option that apparently is the problem of PETSC in Ubuntu 18.04 with OpenMPI 2.1.1. Then, I compiled PETSC  3.8.4 with the new compiled OpenMPI. It passes! 

Then, I compiled preCICE pointing PETSC_DIR to version 3.8.4, but when executing the tests I get the new error.

`1: Test command: /usr/local/bin/mpiexec "-n" "4" "/usr/local/precice/precice-1.6.0/build/testprecice" "--run_test=!@MPI_Ports:!MappingTests/PetRadialBasisFunctionMapping/Parallel/*"
1: Environment variables: 
1:  PRECICE_ROOT=/usr/local/precice/precice-1.6.0
1: Test timeout computed to be: 180
1: /usr/local/precice/precice-1.6.0/build/testprecice: symbol lookup error: /usr/lib/x86_64-linux-gnu/libmpi_cxx.so.20: undefined symbol: opal_uses_threads
1: -------------------------------------------------------
1: Primary job  terminated normally, but 1 process returned
1: a non-zero exit code.. Per user-direction, the job has been aborted.
1: -------------------------------------------------------
1: /usr/local/precice/precice-1.6.0/build/testprecice: symbol lookup error: /usr/lib/x86_64-linux-gnu/libmpi_cxx.so.20: undefined symbol: opal_uses_threads
1: --------------------------------------------------------------------------
1: mpiexec detected that one or more processes exited with non-zero status, thus causing
1: the job to be terminated. The first process to do so was:
1: 
1:   Process name: [[43887,1],0]
1:   Exit code:    127
1: --------------------------------------------------------------------------
1/1 Test #1: precice.Base .....................***Failed    0.05 sec
`
